### PR TITLE
Filebeat Kibana module log fileset

### PR DESCRIPTION
**Pull request by ruflin**
This adds the Kibana module to Filebeat reading the Kibana log. The Kibana log is in JSON format which simplifies reading it but at the same time it provides many fields which are not necessarily know in advance. To solve this issue The data is put under `kibana.log.meta` and the most common known fields are picked into `kibana.log.*` or directly put into the global field like `message`. The fields under `meta` are stored as keyword. This make sure all the meta information around the event is still in the index but prevents potential type conflicts like long vs double.

Additional change:

* Module tests always overwrite the pipeline. This should make development easier.

### CODE REVIEW

**Comment by ruflin** [comment]
@tsg The Kibana log is pretty interesting as it's all in json which is pretty nice. The downside of it is there are lots of fields and I assume in my test logs not all fields that can show up in there. At the moment I'm not sure how we should deal with all these fields as I don't think it will scale to map all of them. I was thinking perhpas we can pick the ones we think are most relevant and put all the others under a namespace and just say it's object type and mapping will be done dynamically?

**Comment by ruflin** [comment]
PR updated with proposed solution for json handling (see PR message). Ready for review.

File: filebeat/filebeat.reference.yml [hunk 1]
```diff
@@ -161,6 +161,16 @@ filebeat.modules:
     #var.paths:
 
 
+#------------------------------- kibana Module -------------------------------
+- module: {{ module }}
```
Comment (kvch) [inline filebeat/filebeat.reference.yml]: Hmmm... This seems to be a problem of the generator. Do you mind changing it manually now. I am opening a PR with the fix in the meantime.

File: filebeat/module/kibana/_meta/config.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+- module: {{ module }}
```
Comment (kvch) [inline filebeat/module/kibana/_meta/config.yml]: Also, this file is problematic.

File: filebeat/modules.d/kibana.yml.disabled [hunk 3]
```diff
@@ -0,0 +1,8 @@
+- module: {{ module }}
```
Comment (kvch) [inline filebeat/modules.d/kibana.yml.disabled]: This one, too.

**Comment by ruflin** [comment]
@kvch I made 2 additional small changes: Added a beta label to the docs and adjusted the path to read logs from.

File: filebeat/module/kibana/log/ingest/pipeline.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
+{
+    "description": "Pipeline for parsing Kibana log logs",
```
Comment (kvch) [inline filebeat/module/kibana/log/ingest/pipeline.json]: Sorry for spotting it quite late, but could you please remove "log" from the description.

**Comment by kvch** [comment]
I tested it locally, works like a charm. After that minor problem is fixed, it's ready to be merged.

**Comment by tsg** [comment]
Would it be possible to add at least a simple Kibana dashboard? So far all FB modules come with at least a dashboard, and I'd like to keep it that way :)

File: filebeat/module/kibana/log/ingest/pipeline.json (hunk 4 above)
Comment (ruflin) [reply filebeat/module/kibana/log/ingest/pipeline.json]: done

**Comment by ruflin** [comment]
Added Dashboards to the meta issue here: https://github.com/elastic/beats/issues/7036

**Pull request merged**